clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -493,12 +509,15 @@ PointerVariableConstraint::mkString(EnvironmentMap &E,
                                     bool ForItype,
                                     bool EmitPointee) {
   std::ostringstream Ss;
-  std::ostringstream Pss;
-  unsigned CaratsToAdd = 0;
+  std::deque<std::string> EndStrs;
+  // This will store stacked array decls to ensure correct order
+  std::stack<std::string> CheckedArrs;
```

Comment:
What's the relationship between `CheckedArrs` and `EndStrs` ? One is a stack and one is a queue, so I am wondering if there is a connection.